### PR TITLE
[Test Modernization]: Add test modernization to Truncate and UnstyledButton

### PR DESCRIPTION
**Pull request by kpinzas-sh**
**WHY are these changes introduced?**

I am updating existing tests for `Truncate` and `UnstyledButton` components to use the new modern framework.

**WHAT is this pull request doing?**

Part of test modernization (https://docs.google.com/spreadsheets/d/1GBuEZbOpVYJLNISK7gL69DU8rCxocn5L5GYaKtPXPbU/edit#gid=1498187033), updating tests using {mountWithAppProvider} from 'test-utilities/legacy' to {mountWithApp} from 'test-utilities'.

### CODE REVIEW

File: src/components/UnstyledButton/tests/UnstyledButton.test.tsx [hunk 1]
```diff
@@ -95,202 +95,232 @@ describe('<Button />', () => {
...
+      expect(button).toContainReactComponent(UnstyledLink, {
+        external: undefined,
+      });
     });
 
     it('is not passed when `url` is missing', () => {
-      const button = mountWithAppProvider(<UnstyledButton external />);
-      expect(button.find('button').prop('external')).toBeUndefined();
+      const button = mountWithApp(<UnstyledButton external />);
+      expect(button).toContainReactComponent('button');
```
Comment (BPScott) [inline src/components/UnstyledButton/tests/UnstyledButton.test.tsx:133]: This should test for the absence of `external`

File: src/components/UnstyledButton/tests/UnstyledButton.test.tsx [hunk 2]
```diff
@@ -95,202 +95,232 @@ describe('<Button />', () => {
...
     });
 
     it('is not passed when `url + disabled`', () => {
-      const button = mountWithAppProvider(
+      const button = mountWithApp(
         <UnstyledButton url={mockUrl} external disabled />,
       );
-      expect(button.find('a').prop('external')).toBeUndefined();
+      expect(button).toContainReactComponent('a');
```
Comment (BPScott) [inline src/components/UnstyledButton/tests/UnstyledButton.test.tsx:140]: This should test for the absence of `external`

**Review by BPScott — APPROVED** [review]
Two little thinks noted inline.  Other than that all looks good to me!

File: src/components/UnstyledButton/tests/UnstyledButton.test.tsx (hunk 1 above)
Comment (kpinzas-sh) [reply src/components/UnstyledButton/tests/UnstyledButton.test.tsx:133]: I updated the PR with the idea i had in mind, let me know what do you think @BPScott

**Pull request merged**